macros/zah.aegi-color-track.lua

```diff
@@ -67,29 +67,9 @@ showDialog = function(macro)
   end
 end
 
--- Main function
-function colortrack(subtitles, selected_lines, active_line)
-  line = subtitles[selected_lines[1]]
-  -- Start gui
-  local res = showDialog("main")
-  if not (res) then
-    return aegisub.cancel()
-  end
-
-  tmp = aegisub.decode_path("?temp")..pathsep.."aegisub-color-tracking"
-
--- Delete old temp files
--- While the script is still running the pixel.png's can't be deleted, because they're considered open.
-  j = 1
-  while (os.remove(tmp..pathsep.."pixel"..j..".png")) ~= nil do
-    os.remove(tmp..pathsep.."frame"..j..".png")
-    os.remove(tmp..pathsep.."pixel"..j..".png")
-    j=j+1
-  end
-
--- Calculate frame perfect times for trimming
-  local starterMS = subtitles[selected_lines[1]].start_time
-  local enderMS = subtitles[selected_lines[1]].end_time
+local function getTimes(line)
+  local starterMS = line.start_time
+  local enderMS = line.end_time
   local startframe = aegisub.frame_from_ms(starterMS)
   local endframe = aegisub.frame_from_ms(enderMS)
```

Comment:
Since frame_from_ms use "[START](https://github.com/Ristellise/AegisubDC/blob/master/src/auto4_lua.cpp#L166)" time, you need to have this code to "balance" the time
local endframe = aegisub..frame_from_ms(enderMS) - 1

Here is the function [FrameAtTime](https://github.com/Aegisub/Aegisub/blob/6f546951b4f004da16ce19ba638bf3eedefb9f31/libaegisub/common/vfr.cpp#L205-L231) that is been called